include/nudb/native_file.hpp

```diff
@@ -8,68 +8,28 @@
 #ifndef NUDB_NATIVE_FILE_HPP
 #define NUDB_NATIVE_FILE_HPP
 
-#include <nudb/error.hpp>
-#include <nudb/file.hpp>
-#include <nudb/posix_file.hpp>
-#include <nudb/win32_file.hpp>
-#include <string>
 
-namespace nudb {
-
-/** A native file handle.
+#if ! GENERATING_DOCS
 
-    This type is set to the appropriate platform-specific
-    implementation to meet the file wrapper requirements.
-*/
-using native_file =
 #ifdef _MSC_VER
-    win32_file;
+#include <nudb/win32_file.hpp>
 #else
-    posix_file;
+#include <nudb/posix_file.hpp>
 #endif
```

Comment:
With this change, doxygen will only see one of the includes. That was not the original intent, its why native_file.hpp includes both of those files. The macros have also been carefully designed so that users can override the behavior of which file gets included. For example if you defined NUDB_POSIX_FILE=1 and NUDB_WIN32_FILE=1 you will get both of them compiled in. Or you can manually choose the one that you want. This change does not appear related to documentation.